Report/sections/3_implementation/expression_type_checking.tex

```diff
@@ -0,0 +1,68 @@
+\subsubsection*{Expression type checking} \label{sec:ExpressionTypeChecking}
+To illustrate an example of how type checking is implemented in \dazel{}, we will now describe the \textbf{ExpressionTypeChecker}.
+Note that we do not exclusively perform type checking on expressions, however this example sufficiently highlights our approach to type checking.
+
+
+Once the type checker visits an assignment node, it initializes an instance of the \textbf{ExpressionTypeChecker} whose responsibility is to traverse the expression and verify that the expression is legal by implementing the \textbf{IExpressionVisitor} interface. 
```

Comment:
```suggestion
Once the \textbf{TypeChecker} visits an assignment node, it initializes an instance of the \textbf{ExpressionTypeChecker}. The \textbf{ExpressionTypeChecker} is a visitor designed to traverse the nodes of an expression. While traversing, it verifies that the expression is legal.
```

Reply:
I think this makes it clearer that the ExpressionTypeChecker is not _the_ type checker. I also think it improves the argument by not stating that it verifies that the expression is legal by 'implementing the \textbf{IExpressionVisitor} interface.'